pkg/packagekit/updates.jsx

```diff
@@ -369,76 +400,72 @@ class OsUpdates extends React.Component {
+                Package: (info, packageId, _summary) => {
                     let id_fields = packageId.split(";");
                     packageSummaries[id_fields[0]] = _summary;
                     updates[packageId] = {name: id_fields[0], version: id_fields[1], security: info == PK_INFO_ENUM_SECURITY};
                     if (id_fields[0] == "cockpit-ws")
                         cockpitUpdate = true;
-                });
+                },
 
-                transProxy.addEventListener("ErrorCode", (event, code, details) => {
+                ErrorCode: (code, details) => {
                     this.state.errorMessages.push(details);
                     this.setState({state: "loadError"});
-                });
-
-                transProxy.addEventListener("changed", (event, data) => {
-                    if ("Status" in data) {
-                        let waiting = (data.Status == PK_STATUS_ENUM_WAIT || data.Status == PK_STATUS_ENUM_WAITING_FOR_LOCK);
-                        if (waiting != this.state.waiting) {
-                            // to avoid flicker, we only switch to "locked" after 1s, as we will get a WAIT state
-                            // even if the package db is unlocked
-                            if (waiting) {
-                                this.setState({waiting: true});
-                                window.setTimeout(() => {!this.state.waiting || this.setState({state: "locked"})}, 1000);
-                            } else {
-                                this.setState({state: "loading", waiting: false});
-                            }
-                        }
-                    }
-                });
+                },
 
                 // when GetUpdates() finished, get the details for all packages
-                transProxy.addEventListener("Finished", () => {
+                Finished: () => {
                     var pkg_ids = Object.keys(updates);
                     if (pkg_ids.length) {
                         this.setState({updates: updates, cockpitUpdate: cockpitUpdate});
                         this.loadUpdateDetails(pkg_ids);
                     } else {
                         this.setState({state: "uptodate"});
                     }
-                });
+                },
+
+            },  // end pkTransaction signalHandlers
+
+            notify => {
+                if ("Status" in notify) {
+                    let waiting = (notify.Status == PK_STATUS_ENUM_WAIT || notify.Status == PK_STATUS_ENUM_WAITING_FOR_LOCK);
+                    if (waiting != this.state.waiting) {
+                        // to avoid flicker, we only switch to "locked" after 1s, as we will get a WAIT state
+                        // even if the package db is unlocked
+                        if (waiting) {
+                            this.setState({waiting: true});
+                            window.setTimeout(() => {!this.state.waiting || this.setState({state: "locked"})}, 1000);
```

Comment:
This timeout is somewhere in the middle... on purpose? The others were `500` and `5000` ms.